### PR TITLE
[Fix](partial-update) Fix partial update fail when the datetime default value is 'current_time'

### PR DESCRIPTION
**Pull request by Yukang-Lian**
Problem: When importing data that includes datetime with a default value of current time for partial column updates, the import fails.

Reason: Partial column updates do not handle the logic for datetime default values.

Solution: During partial column updates, when the default value is set to current time, read the current time from the runtime state and write it into the data.

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by xiaokang** [comment]
@Yukang-Lian test current_date

File: be/src/olap/rowset/segment_v2/segment_writer.cpp [hunk 1]
```diff
@@ -693,7 +695,19 @@ Status SegmentWriter::fill_missing_columns(vectorized::MutableColumns& mutable_f
         for (auto i = 0; i < cids_missing.size(); ++i) {
             const auto& column = _tablet_schema->column(cids_missing[i]);
             if (column.has_default_value()) {
-                auto default_value = _tablet_schema->column(cids_missing[i]).default_value();
+                std::string default_value;
+                if (UNLIKELY(_tablet_schema->column(cids_missing[i]).type() ==
+                                     FieldType::OLAP_FIELD_TYPE_DATETIMEV2 &&
+                             _tablet_schema->column(cids_missing[i])
+                                             .default_value()
+                                             .find("CURRENT_TIMESTAMP") != std::string::npos)) {
```
Comment (zhannngchen) [inline be/src/olap/rowset/segment_v2/segment_writer.cpp:703]: should we consider to compare ignore case?

File: be/src/olap/rowset/segment_v2/vertical_segment_writer.cpp [hunk 2]
```diff
@@ -620,7 +623,19 @@ Status VerticalSegmentWriter::_fill_missing_columns(
         for (auto i = 0; i < missing_cids.size(); ++i) {
             const auto& column = _tablet_schema->column(missing_cids[i]);
             if (column.has_default_value()) {
-                auto default_value = _tablet_schema->column(missing_cids[i]).default_value();
+                std::string default_value;
+                if (UNLIKELY(_tablet_schema->column(missing_cids[i]).type() ==
+                                     FieldType::OLAP_FIELD_TYPE_DATETIMEV2 &&
+                             _tablet_schema->column(missing_cids[i])
+                                             .default_value()
+                                             .find("CURRENT_TIMESTAMP") != std::string::npos)) {
```
Comment (zhannngchen) [inline be/src/olap/rowset/segment_v2/vertical_segment_writer.cpp:631]: ditto

**Comment by Yukang-Lian** [comment]
run buildall

**Review by zhannngchen — DISMISSED** [review]
LGTM

**Comment by Yukang-Lian** [comment]
run buildall

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**